extensions/wikia/Email/EmailController.class.php

```diff
@@ -396,6 +406,80 @@ protected function getSocialMessages() {
 		];
 	}
 
+	private function generateMobileApplicationBadges() {
+		$mobileApplicationsLinks = $this->generateMobileApplicationLinks();
+		$hasMobileApplicationBadges = count($mobileApplicationsLinks) > 0;
+		$badges = [
+			'hasMobileApplicationBadges' => $hasMobileApplicationBadges
+		];
+
+		if ( $hasMobileApplicationBadges ) {
+			if ( $mobileApplicationsLinks['android'] ) {
+				$badges['android'] = [
+					'link' => $mobileApplicationsLinks['android'],
+					'src' => ImageHelper::getFileUrl(
+						$this->getMessage( 'emailext-mobile-application-android-badge' )->text() )
+				];
+			}
+			if ( $mobileApplicationsLinks['ios'] ) {
+				$badges['ios'] = [
+					'link' => $mobileApplicationsLinks['ios'],
+					'src' => ImageHelper::getFileUrl(
+						$this->getMessage( 'emailext-mobile-application-ios-badge' )->text() )
+				];
+			}
+		}
+
+		return $badges;
+	}
+
+	private function generateMobileApplicationLinks() {
+		$result = [];
+
+		$response = $this->fetchMobileApplicationsDetails();
+		$siteId = $this->wg->CityId;
+
+		if ( $this->applicationsExistFor($siteId, $response ) ) {
```

Comment:
It might be nice to reverse this and return early if it evaluates to false. That would help pair down the nested if/forloops. Eg:

```php
if ( !$this->applicationsExistFor( $siteId, $response ) ) {
    return result;
}